### ISSUE TITLE
Disappearing scatter point edges in FacetGrid scatter plot.

### ISSUE THREAD

**Issue by mdickinson** [mwaskom/seaborn#519]
When doing a scatter plot in a `FacetGrid`, the scatter point edges show up for me randomly around 50% of the time.  I have `PYTHONHASHSEED=random` set, so I suspect that this is a result of a dictionary-order dependence somewhere in either the matplotlib or the Seaborn code.  (I'm aware that this could very well turn out to be a matplotlib bug, but so far I've been unable to reproduce the effect with just matplotlib.)

Here's a minimal working example, cut down from something more real:

```python
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

df = pd.DataFrame(
    [
        [1, 2, "class A"],
        [4, 5, "class B"],
        [7, 8, "class B"],
        [1.05, 2.05, "class A"],
    ],
    columns=['A', 'B', 'C'],
)

g = sns.FacetGrid(df, hue="C", size=6)
g.map(plt.scatter, "A", "B", s=200, linewidth=1.0, edgecolor="black")
g.add_legend()
plt.show()
```

When I run this code from the command line with `PYTHONHASHSEED=random python plot_bug.py`, I see one of two possible results, more-or-less at random.  Sometimes the individual scatter points have a black border, as expected ...
![screen shot 2015-04-21 at 15 32 53](https://cloud.githubusercontent.com/assets/662003/7254562/c92251b8-e83b-11e4-9b7e-1dc0994f2680.png)
... and sometimes they don't, which makes the individual points hard to distinguish from each other ...
![screen shot 2015-04-21 at 15 33 08](https://cloud.githubusercontent.com/assets/662003/7254569/dd11b9ca-e83b-11e4-95c9-6aa4656f6c6f.png)

[Version information: Python 2.7.6, matplotlib 1.4.3, Seaborn 0.5.1, Pandas 0.16, OS X 10.9.5]

**Comment by mwaskom** [comment]
When you said you can't reproduce it with "just matplotlib" do you it's specifically dependent on using `FacetGrid` or can you make it happen calling `plt.scatter` directly when seaborn is imported?

**Comment by mdickinson** [comment]
The former.  I tried to reproduce with a simple `plt.scatter` outside the `FacetGrid`, but couldn't.  (Though I haven't spent a lot of time trying variations.)  I didn't remove the `seaborn` import at any point.

**Comment by mdickinson** [comment]
I should also have said that without `PYTHONHASHSEED=random`, I see the expected results every time.  So it looks a lot like a dependence on dictionary order somewhere.

**Comment by mwaskom** [comment]
You could try introspecting the matplotlib objects after plotting to determine if the linewidth attribute is set there properly, could help narrow down whether the problem is on the seaborn side or the drawing side.

**Comment by mwaskom** [comment]
Also just to be precise is it specifically dependent on using a `hue` variable in the `FacetGrid`?

**Comment by mdickinson** [comment]
With a bit more effort, I did manage to reduce it to pure matplotlib, so it looks as though this is a matplotlib bug.  I'll close this issue and open a new one on the matplotlib tracker.  For reference, my minimal failing example now looks like this:

```python
import matplotlib.pyplot as plt
plt.scatter([1, 4, 7], [2, 5, 8], s=200, linewidth=1.0,
            edgecolor='black', color=(0.333, 0.659, 0.408), label='class B')
plt.show()
```

I apologise for the bogus bug report.

**Comment by mwaskom** [comment]
No worries -- weird bug!

**Comment by mdickinson** [comment]
Just to follow up: this is matplotlib/matplotlib#4083.  It's fixed in matplotlib master, and the fix should show up in version 1.5.0 of matplotlib.
